src/lib/support/CHIPMem-Malloc.cpp

```diff
@@ -103,15 +121,27 @@ void * MemoryCalloc(size_t num, size_t size)
 void * MemoryRealloc(void * p, size_t size)
 {
     VERIFY_INITIALIZED();
+    VERIFY_POINTER(p);
```

Comment:
Are these needed? realloc/free don't check themselves in debug mode?

Reply:
The dmalloc implementations do, but I wanted to have the call to MemoryDebugCheckPointer() available for other configurations.